src/components/structures/RoomStatusBar.tsx

```diff
@@ -93,13 +94,15 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
             syncStateData: this.context.getSyncStateData(),
             unsentMessages: getUnsentMessages(this.props.room),
             isResending: false,
+            timelineNeedsRefresh: this.props.room.getTimelineNeedsRefresh(),
         };
     }
 
     public componentDidMount(): void {
         const client = this.context;
         client.on("sync", this.onSyncStateChange);
         client.on("Room.localEchoUpdated", this.onRoomLocalEchoUpdated);
+        client.on("Room.historyImportedWithinTimeline", this.onRoomHistoryImportedWithinTimeline);
```

Comment:
Please add the context type definition around line L87 as string emitter keys are no longer allowed in TypedEventEmitter, TS thinks `this.context` is `any` here though

Reply:
Can we also not listen on the room directly, re-emitters suck

Reply:
> Can we also not listen on the room directly, re-emitters suck

So I don't guess, what should we do? 🙇

Reply:
`this.props.room.on(...);` but with componentDidUpdate comparing this.props.room with prevProps.room to remove old listener and setup a new one. This is all a lot nicer in hooks where its a one-liner.

Reply:
> `this.props.room.on(...);` but with componentDidUpdate comparing this.props.room with prevProps.room to remove old listener and setup a new one. This is all a lot nicer in hooks where its a one-liner.

Updated 👍


> Please add the context type definition around line L87 as string emitter keys are no longer allowed in TypedEventEmitter

I'm not sure what to do here exactly. I've stopped using string keys. Is there anything further?

Reply:
Conversation continued at https://github.com/matrix-org/matrix-react-sdk/pull/8354#discussion_r852587897